src/ImageProcessor.Web/Caching/DiskCache.cs

```diff
@@ -211,64 +211,62 @@ public override Task TrimCacheAsync()
 
             this.ScheduleCacheTrimmer(token =>
             {
-                string rootDirectory = Path.GetDirectoryName(this.CachedPath);
-
+                var rootDirectory = Path.GetDirectoryName(this.CachedPath);
                 if (rootDirectory != null)
                 {
-                    // Jump up to the parent branch to clean through the cache.
-                    // UNC folders can throw exceptions if the file doesn't exist.
-                    IEnumerable<string> directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse();
-
+                    // Jump up to the parent branch to clean through the cache
+                    // UNC folders can throw exceptions if the file doesn't exist
+                    var directories = SafeEnumerateDirectories(validatedAbsoluteCachePath).Reverse().ToList();
                     foreach (string directory in directories)
                     {
-                        if (!Directory.Exists(directory))
-                        {
-                            continue;
-                        }
-
```

Comment:
I've removed this check, as the directory could still be deleted between now and enumerating the files below.